### DISCUSSION TITLE
Weird font discrepancy between terminal font and oh-my-posh.

### DISCUSSION THREAD

**Discussion by Jas-SinghFSU** [kovidgoyal/kitty#7394] [Q&A] [answered]
Firstly, I'm not even sure that this is a kitty issue but it doesn't seem to happen in Wezterm (from which I'm migrating to kitty :D). 

But for some reason, the font style (Medium) isn't being applied properly to the text in oh-my-posh. In the screenshot you can see the text in oh-my-posh and that same text I manually typed right below - which renders perfectly.

Here is the screenshot:
![Screenshot](https://github.com/kovidgoyal/kitty/assets/21210205/391f8b7d-d9d8-4f02-bcc5-6a4fef034740)

Here is my configuration for kitty:
```ini
# Set the default shell program
shell /usr/bin/zsh

# Set the color scheme
# You would need to define the "Catppuccin Mocha" color scheme in a separate theme file
# or manually set each color in this config file if it's not included with Kitty by default.
include ~/.config/kitty/catppuccin-mocha.conf

# Set the tab bar to be at the bottom
tab_position bottom
tab_bar_edge                bottom
tab_bar_style               powerline
tab_powerline_style         slanted
# tab_title_template          {title}{' :{}:'.format(num_windows) if num_windows > 1 else ''}

cursor_blink_interval 0

# Hide the tab bar when only one tab is open
hide_tab_bar_when_only_one_tab yes

# Set the font and font weight
font_family JetBrainsMono NFM Medium
bold_font JetBrainsMono NFM Bold
bold_italic_font JetBrainsMono NFM Bold Italic
italic_font JetBrainsMono NFM Italic
font_size 12
adjust_column_width 90%

# Uncomment the following if you are using wayland; Kitty is known to autodetect Wayland, though.
# wayland_titlebar_side left

# Kitty does not use FreeType or a concept of cell width in the same way WezTerm does
# The following options are not directly translatable to Kitty:
# freetype_render_target: Kitty uses its own rendering method
# cell_width: Kitty does not have this setting as it auto-detects glyph width

# To disable missing glyph warnings
# Kitty doesn't seem to have a direct equivalent for this setting
# vim:ft=kitty

```

Is there a configuration that I'm missing or is this just expected behavior? Really impressed by Kitty's performance btw, its wonderful!

**Comment by kovidgoyal** [comment]
Get the font family names from 

kitty +list-fonts

In kitty dont use patched fonts for nerd font symbols, see the kitty FAQ about it. 

In any terminal, there are basically four fonts uses for text, normal, bold, italic and bold+italic. You can run kitty with the --debug-font-fallback flag to see exactly what fonts it is using.

**Reply by Jas-SinghFSU** [reply]
Interesting, I downloaded the Symbols as stated in the FAQ and changed my font to a standard JetBrains font as stated but it still renders fonts the same way as stated in the main post.

Here's my log output when I run kitty. The fonts are indeed correct (no longer using patched fonts either).
```
[117 00:50:11.194175] normal face: /usr/share/fonts/TTF/JetBrainsMono-Medium.ttf:0
[117 00:50:11.194182] bold face: /usr/share/fonts/TTF/JetBrainsMono-Bold.ttf:0
[117 00:50:11.194187] italic face: /usr/share/fonts/TTF/JetBrainsMono-Italic.ttf:0
[117 00:50:11.194192] bi face: /usr/share/fonts/TTF/JetBrainsMono-BoldItalic.ttf:0
```

Edit: Here's my config now and the `kitty +list-fonts` lists this font as well.
```
font_family JetBrains Mono Medium
bold_font JetBrains Mono Bold
bold_italic_font JetBrains Mono Bold Italic
italic_font JetBrains Mono Italic
font_size 12
adjust_column_width 90%
```

**Reply by kovidgoyal** [reply, marked as answer]
If the fonts as reported by --debug-font-fallback are correct then that's it as far as kitty is concerned. I dont really see the issue in your screneshot but assuming you mean some text is bold and some is not, that will be coming from styling your shell is applying when drawing the prompt. Control it there. 

Also note that text can appear to have different weights in light on dark and dark on light. See the kitty text_composition_strategy option for how you can control the gamma blending curves to adjust that appearance.

**Reply by Jas-SinghFSU** [reply]
Awesome, thank you for the guidance! I will look there as you suggested.

All-in-all its a very minor issue. For reference here's how Wezterm looks where the bold font shows appropriately in the OhMyPosh plugin.

![image](https://github.com/kovidgoyal/kitty/assets/21210205/9f34124e-7296-4b51-8ba7-29af23d76c54)

**Reply by Jas-SinghFSU** [reply]
It seems that achieving the appropriate values for text_composition_strategy did the trick. Greatly appreciate your help!
